python/dgl/heterograph.py

```diff
@@ -1257,7 +1257,78 @@ def batch_num_nodes(self, ntype=None):
         return self._batch_num_nodes[ntype]
 
     def set_batch_num_nodes(self, val):
-        """TBD"""
+        """Manually set the number of nodes for each graph in the batch with the specified node
+        type.
+
+        Parameters
+        ----------
+        val : Tensor or Mapping[str, Tensor]
+            The dictionary storing number of nodes for each graph in the batch for all node types.
+            If the graph has only one node type, ``val`` can also be a single array indicating the
+            number of nodes per graph in the batch.
+
+        Notes
+        -----
+        This API is always used together with ``set_batch_num_edges`` to specify batching
+        information of a graph.
+
+        Examples
+        --------
+
+        The following example uses PyTorch backend.
+
+        >>> import dgl
+        >>> import torch
+
+        Create a homogeneous graph.
+        >>> g = dgl.graph(([0, 1, 2, 3, 4, 5], [1, 2, 0, 4, 5, 3]))
+
+        Manually set batch information
+        >>> g.set_batch_num_nodes(torch.tensor([3, 3])
```

Comment:
If there are edges linking nodes from two specified subgraphs, what will the `unbatch` do?

Reply:
We do not handle this case.

Reply:
Then what is the behavior, will `unbatch` ignore those edges?

A follow up question: if `set_batch_num_nodes` and `set_batch_num_edges` conflict with each other, how does it affect `unbatch` and other related APIs' behavior?

I suggest we either add an API to check that, or add related details in the doc of these two new APIs to warn users.

Reply:
User must guarantee that there will be no invalid (cross-graph) links, but checking validity is too costly and I don't think it necessary to add an API to do so.
I prefer mentioning the behaviour in the docstring.

Reply:
Then would you mind to add a Note/Warning style block in the docstring? 

This note:

- defines *invalid input*, i.e. cross-subgraph links;
- informs users of the potential risks (if called by other APIs like `unbatch`).

Reply:
Done.